### ISSUE TITLE
nginx.conf.sample from 2.4.6-p1 causes problems with running app continer

### ISSUE THREAD

**Issue by redo-interactive** [markshust/docker-magento#958]
**Description**
While installing fresh vanilla magento 2.4.6-p1 app container is off. With error that in nginx.conf in line 50 there is a problem with php-fpm. There are 3 lines with php-fpm. 

When exchanging content from nginx.conf.sample from 2.4.6 it is working fine. 

**Steps To Reproduce**
1. Install fresh version of 2.4.6-p1
2. Check running containers, app should be off. 

**Expected Result**
Application should be working fine, site magento.test should be accessible

**Actual Result**
App container is off with an error related to /var/www/html/nginx.conf. Application isn't accessible.

**Comment by redo-interactive** [comment]
@markshust yes, I have just done that and the project on 2.4.6-p1 works as it should.

**Comment by janmyszkier** [comment]
just small note on this:
`2.4.6-p1` comes with
```
        fastcgi_pass   php-fpm:9000;
```
where `2.4.6`
```
        fastcgi_pass   fastcgi_backend;
```
and conf from 2.4.6 works because the nginx image provides `fastcgi_backend` in: 
https://github.com/markshust/docker-magento/blob/master/images/nginx/1.18/conf/default.conf#L1

but nothing provides `php-fpm` backend definition.

This COULD be scripted to check whether the backend under the name from nginx.conf.sample exists (with bash) and added as a new script  like `check-requirements` within the bin dir just like other commands  BUT the nginx.sample should never be used for real setup, because you should just copy the sample and modify it to your need.
SO I guess, it would be probably better for the repo to provide it's own nginx.conf and copy it to the project directory as a part of the setup.

**Comment by piotrkwiecinski** [comment]
https://experienceleague.adobe.com/docs/commerce-operations/release/notes/security-patches/2-4-6-p1.html this is a known issue:

> The nginx.sample file was inadvertently updated with a change that modifies the value of fastcgi_pass from fastcgi_backend to php-fpm:9000. This change can be safely reverted or ignored.

I would expect it'll be reverted in next update.

**Comment by piotrkwiecinski** [comment]
Looks like it was fixed in https://experienceleague.adobe.com/docs/commerce-operations/release/notes/security-patches/2-4-6-p2.html
@markshust maybe it's worth bumping defaults to p2?

**Comment by markshust** [comment]
This was an upstream bug in Adobe's `nginx.conf.sample` shipped with 2.4.6-p1 only, and Adobe fixed it in 2.4.6-p2 (per their [release notes](https://experienceleague.adobe.com/docs/commerce-operations/release/notes/security-patches/2-4-6-p2.html)). Defaults now target 2.4.9, so this is no longer reachable. Closing — thanks all!
